mlrun/projects/project.py

```diff
@@ -2305,6 +2306,20 @@ def create_remote(self, url, name="origin", branch=None):
         self.spec._source = self.spec.source or url
         self.spec.origin_url = self.spec.origin_url or url
 
+    def is_git_initialized(self):
+        context = self.context
+        git_dir_path = os.path.join(context, ".git")
+
+        if os.path.exists(git_dir_path) and os.path.isdir(git_dir_path):
```

Comment:
check if path does not exist, then log a warning that git was not initialized and we are about to do it. use
```
repo = git.Repo.init(context)
```
otherwise, create the repo from the context like you did below